### PR TITLE
Clean up some hydro stuff in PointwiseFunctions

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Use known types instead of type aliases.
Clean up includes.
Use default frame of hydro tags.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.cpp [hunk 1]
```diff
@@ -18,10 +19,11 @@
...
 template <typename DataType>
-Scalar<DataType> compute_piecewise(
-    const tnsr::I<DataType, 3, Frame::Inertial>& x, const double& inner_radius,
-    const double& outer_radius, const double& inner_value,
-    const double& outer_value) noexcept {
+Scalar<DataType> compute_piecewise(const tnsr::I<DataType, 3>& x,
+                                   const double& inner_radius,
+                                   const double& outer_radius,
+                                   const double& inner_value,
+                                   const double& outer_value) noexcept {
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.cpp:26]: I don't believe there is an advantage to taking these doubles by reference, and AFAIK doing so adds the expense of a pointer dereference... I suggest taking them by value instead.

File: src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp [hunk 2]
```diff
@@ -130,12 +128,12 @@ class CylindricalBlastWave : public MarkAsAnalyticData {
...
-  CylindricalBlastWave(
-      InnerRadius::type inner_radius, OuterRadius::type outer_radius,
-      InnerDensity::type inner_density, OuterDensity::type outer_density,
-      InnerPressure::type inner_pressure, OuterPressure::type outer_pressure,
-      MagneticField::type magnetic_field, AdiabaticIndex::type adiabatic_index,
-      const OptionContext& context = {});
+  CylindricalBlastWave(double inner_radius, double outer_radius,
+                       double inner_density, double outer_density,
+                       double inner_pressure, double outer_pressure,
+                       std::array<double, 3> magnetic_field,
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp:134]: I believe we usually take non-fundamental types by const-ref? (I suppose there is an argument to be made that a std:array<double, 3> is a small type and should be ok?). For consistency I would suggest changing to a const-ref here and elsewhere throughout the PR.

File: src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp [hunk 3]
```diff
@@ -220,24 +216,17 @@ class CylindricalBlastWave : public MarkAsAnalyticData {
...
+  double inner_radius_ = std::numeric_limits<double>::signaling_NaN();
+  double outer_radius_ = std::numeric_limits<double>::signaling_NaN();
+  double inner_density_ = std::numeric_limits<double>::signaling_NaN();
+  double outer_density_ = std::numeric_limits<double>::signaling_NaN();
+  double inner_pressure_ = std::numeric_limits<double>::signaling_NaN();
+  double outer_pressure_ = std::numeric_limits<double>::signaling_NaN();
+  std::array<double, 3> magnetic_field_{
+      {std::numeric_limits<double>::signaling_NaN(),
+       std::numeric_limits<double>::signaling_NaN(),
+       std::numeric_limits<double>::signaling_NaN()}};
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp:228]: Should be cleaner as `auto blah = make_array<3>(nan)`, also applies throughout the PR.

**Review by fmahebert — APPROVED** [review]
LGTM, but here are three nitpicks for even cleaner cleaning.

Edit to clarify: suggestions are non binding

**Review by nilsdeppe — APPROVED** [review]
I agree with @fmahebert that some things could be improved a little bit more, but also think this is a very nice improvement and don't want to hold it up :) Either way, here's a seal

image

**Comment by carmaza** [comment]
Any particular reason why you didn't touch the NewtonianEuler stuff?

**Comment by kidder** [comment]
@carmaza I chose the files I touched based on wanting to remove `item_type` and `::type` when the type was known in files in `PointwiseFunctions` and that included `Hydro/Tags.hpp`.

**Comment by carmaza** [comment]
Ah OK. `AnalyticSolutions/NewtonianEuler` has some `Frame::Inertial` hard-coded too, so feel free to remove those as well or leave them to me for a future PR. I'm sure @fmahebert will love to be consistent here :P

**Comment by fmahebert** [comment]
Oh yes 😛 

I'm happy to make the changes I suggested above (unless Larry prefers to make them here/now), and the changes to NewtonianEuler that Cris mentioned, in an upcoming cleanup PR...

File: src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp (hunk 3 above)
Comment (kidder) [reply src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp:228]: I chose this way (i.e explicit declaration of a size 3 array over make_array) to make these declarations consistent to eliminate including make_arry; but I don't have a strong preference

**Comment by kidder** [comment]
okay, then I will let @fmahebert make a follow up PR as I have other things to do...

File: src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp (hunk 3 above)
Comment (fmahebert) [reply src/PointwiseFunctions/AnalyticData/GrMhd/CylindricalBlastWave.hpp:228]: Ah, okay; thanks for clarifying

**Pull request merged**